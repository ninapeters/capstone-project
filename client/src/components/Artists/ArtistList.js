import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import Artist from './Artist'

ArtistList.propTypes = {
  artists: PropTypes.array,
  updateTrack: PropTypes.func,
  toggleCurrentSongId: PropTypes.func.isRequired,
  isSongPlaying: PropTypes.bool.isRequired,
  currentSongId: PropTypes.string,
  saveSong: PropTypes.func.isRequired,
  savedSongs: PropTypes.array,
}

export default function ArtistList({
  artists,
  updateTrack,
  stopPlayingSong,
  toggleCurrentSongId,
  isSongPlaying,
  currentSongId,
  saveSong,
  savedSongs,
}) {
  return (
    <ListStyled>
      {artists?.map(({ artist, songTitle, image, id }) => (
        <Artist
          key={id}
          artist={artist}
          songTitle={songTitle}
          image={image}
          id={id}
          updateTrack={updateTrack}
          stopPlayingSong={stopPlayingSong}
          toggleCurrentSongId={toggleCurrentSongId}
          isSongPlaying={isSongPlaying}
          currentSongId={currentSongId}
          saveSong={saveSong}
          savedSongs={savedSongs}
        />
      ))}
    </ListStyled>
  )
}

const ListStyled = styled.ul`
  display: flex;
  flex-wrap: nowrap;
  height: 100vh;
  margin: 0;
  overflow-x: scroll;
  padding: 0;
  scroll-snap-type: x mandatory;
  scrollbar-width: none;
  &::-webkit-scrollbar {
    display: none;
  }
`
