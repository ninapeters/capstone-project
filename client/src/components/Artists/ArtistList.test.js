import { render } from '@testing-library/react'
import ArtistList from './ArtistList'

describe('SavedSongList', () => {
  it('renders correctly', () => {
    const { container } = render(
      <ArtistList
        deleteSavedSong={() => {}}
        artists={[
          {
            artist: 'Unprocessed',
            songTitle: 'Real',
            songSnippet: 'https://test/',
            id: 'a',
          },
          {
            artist: 'Dance Gavin Dance',
            songTitle: 'We Own The Night',
            songSnippet: 'https://test/',
            id: 'b',
          },
        ]}
        updateTrack={() => {}}
        toggleCurrentSongId={() => {}}
        isSongPlaying={false}
        currentSongId="a"
        saveSong={() => {}}
        savedSongs={[]}
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
