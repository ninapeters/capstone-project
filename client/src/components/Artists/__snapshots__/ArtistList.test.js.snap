// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedSongList renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  height: 100vh;
  margin: 0;
  overflow-x: scroll;
  padding: 0;
  -webkit-scroll-snap-type: x mandatory;
  -moz-scroll-snap-type: x mandatory;
  -ms-scroll-snap-type: x mandatory;
  scroll-snap-type: x mandatory;
  -webkit-scrollbar-width: none;
  -moz-scrollbar-width: none;
  -ms-scrollbar-width: none;
  scrollbar-width: none;
}

.c0::-webkit-scrollbar {
  display: none;
}

<ul
  class="c0"
/>
`;
